Fail to checkout opam-repository

Interaction:
```
$ dune-release opam submit
Submitting _build/ANSITerminal.0.8.1
From https://github.com/ocaml/opam-repository
 * branch                  master     -> FETCH_HEAD
dune-release: [ERROR] ['git' '--git-dir' '.git' '--work-tree' './' 'checkout'
                      '--quiet' '-b'
                       'release-ANSITerminal-0.8.1'
                      'b1ac8b1e1e4ad5da0a6e09713f413c07eac6ed52'] exited with
                      code 1
```

```
$ dune-release opam submit -vv
dune-release: [INFO] dune-release 1.0.1 running
dune-release: [EXEC:29493] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:29494] ['git' '--git-dir' '.git' '--work-tree' './'
                            'describe' '--always' 'HEAD']
dune-release: [EXEC:29495] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:29496] ['git' '--git-dir' '.git' '--work-tree' './'
                            'describe' '--always' 'HEAD']
Submitting _build/ANSITerminal.0.8.1
dune-release: [EXEC:29497] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:29498] ['git' '--git-dir' '.git' '--work-tree' './'
                            'describe' '--always' 'HEAD']
dune-release: [INFO] Parsing opam file ANSITerminal.opam
dune-release: [EXEC:29499] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:29500] ['git' '--git-dir' '.git' '--work-tree' './'
                            'describe' '--always' 'HEAD']
dune-release: [EXEC:29501] ['git' 'rev-parse' '--git-dir']
dune-release: [EXEC:29502] ['git' '--git-dir' '.git' 'fetch'
                            'https://github.com/ocaml/opam-repository.git'
                            'master']
From https://github.com/ocaml/opam-repository
 * branch                  master     -> FETCH_HEAD
dune-release: [EXEC:29510] ['git' '--git-dir' '.git' 'rev-parse'
                            'FETCH_HEAD']
dune-release: [EXEC:29511] ['git' '--git-dir' '.git' 'show-ref' '--verify'
                            '--quiet'
                            'refs/heads/release-ANSITerminal-0.8.1']
dune-release: [EXEC:29512] ['git' '--git-dir' '.git' '--work-tree' './'
                            'checkout' '--quiet' '-b'
                            'release-ANSITerminal-0.8.1'
                            'b1ac8b1e1e4ad5da0a6e09713f413c07eac6ed52']
dune-release: [ERROR] ['git' '--git-dir' '.git' '--work-tree' './' 'checkout'
                      '--quiet' '-b'
                       'release-ANSITerminal-0.8.1'
                      'b1ac8b1e1e4ad5da0a6e09713f413c07eac6ed52'] exited with
                      code 1
```

--- a/lib/opam.ml
+++ b/lib/opam.ml
@@ -80,8 +80,13 @@ let prepare ~dry_run ?msg ~local_repo ~remote_repo ~version names =
     let delete_branch () =
       if not (Vcs.branch_exists ~dry_run:false repo branch) then Ok ()
       else (
-        run Cmd.(git % "checkout" % "master") >>= fun () ->
-        run Cmd.(git % "branch" % "-D" % branch)
+        match run Cmd.(git % "checkout" % "master") with
+        | Ok ()   -> run Cmd.(git % "branch" % "-D" % branch)
+        | Error _ ->
+            let out = OS.Cmd.run_out Cmd.(git % "status") in
+            OS.Cmd.out_lines out >>= fun (out, _) ->
+            R.error_msgf "git checkout in %a failed:\n %s"
+              Fpath.pp local_repo (String.concat ~sep:"\n" out)
       )
     in
     delete_branch () >>= fun () ->

